tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -348,15 +348,21 @@ public async Task ConcurrentJobConfigurationShouldBeRespected()
             .And.WithCronExpression("* * * * *").WithName("Job 2")
             .And.WithCronExpression("* * * * *").WithName("Job 3")
             .And.WithCronExpression("* * * * *").WithName("Job 4")));
+
         var provider = CreateServiceProvider();
 
+        (IDisposable subscriber, IList<ExecutionProgress> events) = RegisterAnExecutionProgressSubscriber(provider);
+
         await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
 
         FakeTimer.Advance(TimeSpan.FromMinutes(1));
-        // Wait 2 instances at the same time
-        (await WaitForJobsOrTimeout(2, TimeSpan.FromMilliseconds(150))).ShouldBeTrue();
-        // But not another instance
-        (await WaitForJobsOrTimeout(1, TimeSpan.FromMilliseconds(50))).ShouldBeFalse();
+        await Task.Delay(TimeSpan.FromMilliseconds(1), CancellationToken);
```

Comment:
For what is the Delay needed - I saw your PR title :D this might be a culprit of flakiness introduced? :D you know - while the other flakiness is gone :D

Reply:
It's the only way I've found to give some leeway so that the Task pump can start processing thing at the lib level and actually perform some scheduling.

But you're right. It's not bulletproof. Let me try another approach.

Reply:
Hrmpf. Fighting against the dreaded "Collection was modified; enumeration operation may not execute.".

I'll sleep over it and come back to this with a fresher pair of eyes.

Reply:
Don’t worry - will do the same :)

Reply:
@linkdotnet Found an approach. What do you think?

Reply:
Yeah I like that!